specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json

```diff
@@ -2605,6 +2605,60 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataFactory/factories/{factoryName}/triggers": {
```

Comment:
Isnt it possible to enable this via query filter parameter on GET collection API? Using $filter. trigger type is a property on the trigger resource, so that will work. Not sure about parent trigger name. triggers cant be modeled as an action and as a resource type. Right now its modeled as one since there is GET/PUT on triggers resource and now you are trying to model a POST on triggers.

Reply:
this is an action on triggers, like a list call to get all triggers(which is also post), this is a query to get triggers with some filters

Reply:
Gaurav, Our get APIs do not support filters, they are simple List APIs

---
In reply to: [327443711](https://github.com/Azure/azure-rest-api-specs/pull/6903#discussion_r327443711) [](ancestors = 327443711)

Reply:
@hvermis - this will be confusing to the users. they can do a GET on triggers and a POST on triggers. So is triggers an entity or an action on an entity. Strong recommendation is to supportsuch scenarios using $filter. You are not helping your customers by adding a new API for every such usecase where they want to get a subset of the result set. Other option will be to call this something else (not triggers). There is a RBAC loop hole also in this. A customer has a custom role which lists triggers/read as a not action. BUt they can still see the trigger by calling this POST action.

Reply:
that is because Sparsh named the API incorrectly, it should be queryTriggers in the factory scope

Reply:
@spagarwa this comment is still not addressed (see hvermis's latest comment)